packages/core/src/transports/utils.ts

```diff
@@ -1,9 +1,67 @@
+import type { IResourceSpans } from '@opentelemetry/otlp-transformer';
+
+import type { EventEvent, ExceptionEvent, LogEvent, MeasurementEvent, TraceEvent } from '..';
+
 import { TransportItemType, transportItemTypeToBodyKey } from './const';
 import type { TransportBody, TransportItem } from './types';
 
-export function getTransportBody(item: TransportItem): TransportBody {
+export function mergeResourceSpans(traces?: TraceEvent, resourceSpans?: IResourceSpans[]): TraceEvent {
+  if (traces === undefined) {
+    return {
+      resourceSpans,
+    };
+  }
+
+  const currentResource = traces.resourceSpans?.[0];
+  if (currentResource === undefined) {
+    return traces;
+  }
+
+  const currentSpans = currentResource?.scopeSpans || [];
+  const newSpans = resourceSpans?.[0]?.scopeSpans || [];
+
   return {
-    [transportItemTypeToBodyKey[item.type]]: item.type === TransportItemType.TRACE ? item.payload : [item.payload],
-    meta: item.meta,
+    ...traces,
+    resourceSpans: [
+      {
+        ...currentResource,
+        scopeSpans: [...currentSpans, ...newSpans],
+      },
+    ],
+  };
+}
+
+export function getTransportBody(item: TransportItem[]): TransportBody {
+  let body: TransportBody = {
+    meta: {},
   };
+
+  if (item[0] !== undefined) {
+    body.meta = item[0].meta;
+  }
+
+  item.forEach((currentItem: TransportItem) => {
+    switch (currentItem.type) {
+      case TransportItemType.LOG:
+      case TransportItemType.EVENT:
+      case TransportItemType.EXCEPTION:
+      case TransportItemType.MEASUREMENT:
+        const bk = transportItemTypeToBodyKey[currentItem.type];
```

Comment:
Q:
do we need the transform function?

We can do 
```ts
const bodyKey = `${currentItem.type}s`

const signals = body[bodyKey] as ...

 body = {
          ...body,
          [bodyKey]: signals === undefined ? [currentItem.payload] : [...signals, currentItem.payload],
        };
```

Reply:
I think I prefer the current way with switch which is more explicit but not very strong opinion